### PR TITLE
`General`: Improve student statistics layout, show learning goals to students again and align colors on course overview

### PR DESCRIPTION
**Pull request by willergeorg**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [ ] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

The current layout of the student statistics view is improbable. @krusche pitched me some ideas about an enhanced design.
These ideas are realized in this PR.
Additionally, the learning goals are currently not displayed to the students. This is the case because the activated route is not able to fetch the course id from the parent. This information is needed for the REST request to the server.
### Description

The following changes have been done:

- **General structure**: The different charts are now aligned below each other. Additionally, every section has its own title and is separated by lines
- **Ordering**: Below the doughnut chart, the Performance in Exercises line chart appears. Below that, the horizontal bar charts are placed
- **Color distribution**: The color legend in the doughnut chart is now moved from the horizontal bar charts to the chart itself. The color legend also ensures translation sensitivity.
- **Chart appearance**: The height of a horizontal bar has been halved
- **Font size**: The font size of the statistical values on the right side of the view have been aligned with the color legend of the charts
- **Color alignment**: The doughnut chart on course overview now provide colors aligned with the reworked color distribution of Artemis.
- **Learning goals**: The learning goals appear in the statistics again.

### Steps for Testing

Prerequisites:
- 1 Course with exercises and participation
- 1 Instructor
- 1 Student
- 1 Programming Exercise with Complaints enabled

1. Log in to Artemis as student
2. Inspect the course overview page. Make sure the color of the doughnut charts are muted
3. Navigate to the student statistics tab
4. Check the new layout. Make sure everything is still readable and nothing is misaligned. Make sure no learning goals are visible
5. Log in as instructor
6. Create a learning goal for a exercise attached to a lecture
7. Log in as student again
8. Navigate to the student statistics tab
9. Make sure the learning goal is visible now
10. Also make sure everything works with the new dark mode
11. If you have suggestions, feel free to share them. Then we can consider them for potential changes

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| `course-statistics.component.ts` | 87.09% | 95.09% |
| `course-card.component.ts` | 88.67% | 96.15% |

### Screenshots

The new colors of the doughnut charts on course overview:
image

First part of the new student's statistics structure:
image

Second part:
image

### CODE REVIEW

**Review by nairaarnez — APPROVED** [review]
Tested on ts3. It works as expected, also in dark mode

**Review by rhubinger — APPROVED** [review]
Tested on TS2. Everything worked as described both in dark and normal mode

**Review by max-bergmann — APPROVED** [review]
Tested on TS3. Everything worked as expected.

File: src/main/webapp/app/overview/course-card.component.ts [hunk 1]
```diff
@@ -45,7 +46,7 @@ export class CourseCardComponent implements OnChanges {
         name: 'vivid',
         selectable: true,
         group: ScaleType.Ordinal,
-        domain: ['#32cd32', '#ff0000'], // colors: green, red
+        domain: [GraphColors.GREEN, GraphColors.RED], // colors: green, red
```
Comment (b-fein) [inline src/main/webapp/app/overview/course-card.component.ts:49]:
```suggestion
        domain: [GraphColors.GREEN, GraphColors.RED],
```

File: src/main/webapp/app/overview/course-statistics/course-statistics.component.html [hunk 2]
```diff
@@ -46,77 +46,101 @@
...
+                        <div class="mt-3">
+                            <a class="btn btn-primary" [routerLink]="['grading-key']" [queryParams]="{ grade: gradeDTO?.gradeName }">
+                                <fa-icon [icon]="faClipboard"></fa-icon>
+                                <span
+                                    >&nbsp;{{
+                                        isBonus
+                                            ? ('artemisApp.exam.examSummary.gradeKeyButtonBonus' | artemisTranslate)
+                                            : ('artemisApp.exam.examSummary.gradeKeyButton' | artemisTranslate)
+                                    }}</span
+                                >
```
Comment (b-fein) [inline src/main/webapp/app/overview/course-statistics/course-statistics.component.html:120]:
```suggestion
                                <span>
                                    {{
                                        isBonus
                                            ? ('artemisApp.exam.examSummary.gradeKeyButtonBonus' | artemisTranslate)
                                            : ('artemisApp.exam.examSummary.gradeKeyButton' | artemisTranslate)
                                    }}
                                </span>
```
Looks like the usual autoformatter problem. I think in most places there is no special non-breaking space between icon and the text. I’m not sure if it is specifically needed here.

Just applying the suggestion will probably make the formatter complain again, it’s probably best to make the change manually.

**Review by b-fein — COMMENTED** [review]
Code looks good with only minor suggestions.

It would also be nice if the single-line change of hard-coded colours to the `GraphColors.{GREEN, RED}` as noted in https://github.com/ls1intum/Artemis/issues/5044#issuecomment-1120300503 could be implemented as part of this PR.

**Review by Dominik-Weinzierl — COMMENTED** [review]
The code looks good to me. You could also think about if `.statistics-legend` is really needed. Instead you could use `fw-bold` ([Link](https://getbootstrap.com/docs/5.0/utilities/text/)) :thinking:

File: src/main/webapp/app/overview/course-card.component.ts (hunk 1 above)
Comment (willergeorg) [reply src/main/webapp/app/overview/course-card.component.ts]: I changed the color implementation in `doughnut-chart.component.ts `accordingly in 37bdc4e.

File: src/main/webapp/app/overview/course-statistics/course-statistics.component.html (hunk 2 above)
Comment (willergeorg) [reply src/main/webapp/app/overview/course-statistics/course-statistics.component.html]: It's always a pleasure for me to get my code reviewed by you but one day we have to evaluate the reasonableness of fighting the autoformatter. 😄 I tried my best to fix the structure manually as you proposed.

**Comment by willergeorg** [comment]
> The code looks good to me. You could also think about if `.statistics-legend` is really needed. Instead you could use `fw-bold` ([Link](https://getbootstrap.com/docs/5.0/utilities/text/)) 🤔

Thanks for mentioning that. I was not aware of the fact that bootstrap provides this class (even though it should be quite obvious that bootstrap has your back there as always). I changed the implementation accordingly. 👍

**Review by Dominik-Weinzierl — APPROVED** [review]
Code looks good to me :tada: thanks for implementing my suggested changes :+1:

**Review by b-fein — APPROVED** [review]
Thanks for implementing the suggestions, code looks good.

**Review by rhubinger — APPROVED** [review]
Tested on TS2. Everything worked, LGTM.

**Review by nairaarnez — APPROVED** [review]
I tested the PR in ts1, everything works as discribed

**Review by jschmautzer — APPROVED** [review]
tested on ts3, worked as expected :)

**Review by bassner — APPROVED** [review]
Code looks good to me.
Tested this on TS5 and it works fine.

I have some remarks that can be solved in a follow-up PR as they were not introduced by this PR :)

**Pull request merged**